papermario/src/code_fa4c0_len_3bf0.c

```diff
@@ -48,19 +143,52 @@ INCLUDE_ASM(code_fa4c0_len_3bf0, DropItemEntityB);
 
 INCLUDE_ASM(code_fa4c0_len_3bf0, RemoveItemEntity);
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, AddBadge);
+s32 AddBadge(script_context* script, s32 initialCall) {
+  s32* ptrReadPos = script->ptrReadPos;
+  s32 itemID = get_variable(script, *ptrReadPos++);
+  s32* ptrNextPos = *ptrReadPos++;
+
+  set_variable(script, ptrNextPos, add_badge(itemID));
+  return 2;
+}
 
 INCLUDE_ASM(code_fa4c0_len_3bf0, RemoveBadge);
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, SetItemPos);
+s32 SetItemPos(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    item_entity* ptrItemEntity;
+    s32 itemEntityIndex;
+    s32 x, y, z;
+
+    itemEntityIndex = get_variable(script, *ptrReadPos++);
+    x = get_variable(script, *ptrReadPos++);
+    y = get_variable(script, *ptrReadPos++);
+    z = get_variable(script, *ptrReadPos++);
+
+    ptrItemEntity = get_item_entity(itemEntityIndex);
+    ptrItemEntity->position[0] = (f32) x;
```

Comment:
These casts shouldn't be necessary, because these fields are already floats so the casts will happen automatically